### DISCUSSION TITLE
Semester 4 Woche 2

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#41] [Student Exchange]
Hallo zusammen,

diese Woche haben wir ein paar mögliche 🚨 Risiken für das Projekt gesammelt.
Unten findet ihr den momentanen Stand der Risikoliste. Für weitere Informationen zu der Liste könnt ihr gerne direkt ins [📄 ASR](https://github.com/Christian-2003/CrInGE/blob/master/docs/ArchitektonischRelevanteAnforderungen.md#11-risiken-und-technische-schulden) schauen.
Wir 🙏 hoffen natürlich, dass keine der Risiken eintritt.

Desweiterem haben wir die verschiedenen Branches des letzten Semesters in den master Branch 🔀 gemerged und uns wieder mit dem Projekt vertraut gemacht. Somit können wir jetzt wieder 💪 effektiv an unserem letzte Woche geplantem Ziel für die Abschluss Demo arbeiten.

Das wäre dann auch schon alles für diese Woche. Im nächsten Blog werden wir euch weiter über das Projekt informieren.

### Risikoliste
| ID | Beschreibung | Wahrscheinlichkeitsklasse | Schadensklasse | Risiko-Score | Minimierungs-Strategie | Indikatoren | Notfallplan | Status  | Verantwortlicher | Datum der letzten Aktualisierung |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 09 | Kritische Echtzeit Performance | 3 | 5 | 15 | Abläufe möglichst effizient und frei von Overhead halten | Tests der Features mit größeren Datenmengen | Entsprechende Features mit Warnungen zu kritischen Größen versehen, sofern kein offensichtliches Verbesserungspotenzial besteht | Offen | Jannik | 15.04.2024 |
| 02 | Unvollständiger Editor | 3 | 4 | 12 | Paralleles Entwickeln der Features sowohl in Editor als auch Engine | Funktionen der Engine sind nicht in dem Editor umgesetzt | Ausführliche Dokumentation der offenen bzw. benötigten Schnittstellen | Offen | Tim | 15.04.2024 |
| 03 | Unvollständige Engine | 2 | 4 | 8 | Paralleles Entwickeln der Features sowohl in Editor als auch Engine | Funktionen des Editor sind nicht in der Engine umgesetzt | Sperren der Features im Editor | Offen | Felix | 15.04.2024 |
| 04 | Fehlerhafte Schnittstelle zwischen Engine und Editor | 2 | 4 | 8 | Klare Dokumentation der benötigten Datenstrukturen | Daten, welche der Editor erstellt, können von der Engine nicht gelesen werden | Parserfunktionen zur Verfügung stellen oder anpassen | In Bearbeitung | Kai | 15.04.2024 |
| 05 | Mangelnde Kommunikation innerhalb des Projektteams | 2 | 4 | 8 | Organisation über Scrumboard und Discord | Teammitglied weiß nicht über die Aufgabenverteilung bescheid oder arbeitet ohne Wissen des restlichen Teams | Krisensitzung nach einer Vorlesung, zu welcher alle Teammitglieder erschienen sind | Reduziert | Christian | 15.04.2024 |
| 08 | Unrealistische Zeit- und Aufwandsabschätzung | 4 | 2 | 8 | Überwachung der Diverenz zwischen geplantem und erfolgtem Arbeitsaufwand via Scrumboard-Statisiken | Aufgaben werden überwiegend deutlich zu komplex oder trivial geschätzt | Rücksprache des Teams mit Scrummaster (Tim) | Offen | Tim | 15.04.2024 |
| 01 | Schlechte Note | 1 | 5 | 5 | Priorisierung der Aufgaben an Benotungsrichtlinien anpassen | Aufgaben, welche Punkte für die Benotung geben, werden nicht fristgerecht abgeschlossen | Auf besonders wichtige und nicht abgeschlossene Aufgaben kurz vor Ende der Frist nochmals persönlich hinweisen | Reduziert | Tim | 15.04.2024 |
| 06 | Unklare Definition des Umfangs und der Funktionen der Abschlussdemo | 1 | 3 | 3 | Detailierte ausformulierte Liste der Ziele | Unklarheit bei Definition der Aufgaben durch Teammitglieder | Reduziert | Face to Face Gespräch | Tim | 15.04.2024 |

**Comment by lukpr0** [comment]
Hallo Cringe-Team
Ich finde gut wie ausführliche eure Risikoanalyse ist. Mir stellt sich aber die Frage, warum ihr die Echtzeit Performance als kritischer betrachtet als die Vollständigkeit eures Projekts (Punkte 2,3,4)?

Bei Punkt 5 ist euch glaube ich ein Fehler beim Riskoscore unterlaufen

**Reply by TimSchnur** [reply]
Danke schonmal für den Hinweis zu Punkt 5.  Dies ist bereits behoben.

Bezüglich deiner Frage: Wenn das Projekt Teilweise unvollständig ist kann dem Problem recht gut gegengewirkt werden. Falls wir allerdings auf Performance Probleme stoßen, könnte das Projekt endgültig scheitern oder müsste stark eingeschränkt werden.

**Comment by PetrusQn** [comment]
Hi Cringe-Team,
ich halte eure Risikoanalyse ebenfalls für sehr gelungen. @lukpr0 hat meiner Meinung nach bereits das Wichtigste erwähnt und ich habe nichts hinzuzufügen.
